packages/client/src/authentication/__helpers__/setupAuthentication.ts

```diff
@@ -8,38 +8,34 @@ const testConfig = {
   environment: mumbaiSandbox,
 };
 
-let auth: Authentication;
+export function setupAuthentication() {
+  let auth: Authentication;
 
-export async function setupAuthentication(): Promise<Authentication> {
-  if (auth) {
-    return auth;
-  }
+  beforeAll(async () => {
+    const wallet = setupTestWallet();
+    auth = new Authentication(testConfig);
+    const address = await wallet.getAddress();
+    const challenge = await auth.generateChallenge(address);
+    const signature = await wallet.signMessage(challenge);
 
-  const wallet = setupTestWallet();
-  auth = new Authentication(testConfig);
-  const address = await wallet.getAddress();
-  const challenge = await auth.generateChallenge(address);
-  const signature = await wallet.signMessage(challenge);
+    await auth.authenticate(address, signature);
```

Comment:
this is executed separately per test file causing race conditions, conflicts with credentials, and API returns "not authenticated" for the tests that were too slow (or too fast). I skipped many tests in this PR and want to fix that in a separate ticket https://avara.height.app/T-6018